studio: further enhancements

tapadipti opened the pull request:
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

duijf commented on content/docs/studio/create-view.md:
```diff
@@ -57,11 +71,20 @@ You will need to specify advanced settings in the following scenarios.
+![](https://static.iterative.ai/img/studio/configure_github.png)
+
+Click on the `Configure` button, and you will be redirected to the GitHub page
+to install and authorize the DVC Studio GitHub app.
```
If there is time: it might also be good to document the steps that users can take on GitHub + add screenshots.

It might also be worthwhile to note that users can only give us access to the repositories they want to use with Studio. They don't have to grant us access to all repositories in their GitHub org if they don't want to.

tapadipti replied:
Thanks @duijf I've added a note regarding limiting the access to only the required repositories. 
Regarding documenting the steps to be taken on GH, I was thinking that since that part is external to us, maybe we should not add details/screenshots for that, coz we don't have control on if/when there's any change in that. What do you think?

tapadipti replied:
Added notes below for limiting access.

duijf replied:
Cool! Thanks :)

Regarding screenshots: I feel like users might benefit from the additional docs more than what this would cost us in terms of maintenance.

While the UI on GitHub's side may change slightly, the main concepts should continue to apply. (If the flow changes in major ways, they will probably announce this somehow). Having the screenshots be slightly outdated won't be the end of the world I think + it's relatively quick to fix after we find out.

Your call though!

tapadipti replied:
Got it. I'll create a follow-up ticket to add more GH instructions/screenshots, coz I think it needs some restructuring - will be good to move this section into a new page, coz the current one is quite long already and has too many screenshots.

shcheklein replied:
now it's a comment here https://github.com/iterative/dvc.org/issues/2515

duijf reviewed (commented):
Thanks for documenting the GitHub app stuff @tapadipti! Looks good to me!

I have added two suggestions for things that we could mention in addition.

tapadipti commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+# Additional Settings for a View
```
@shcheklein This is the page where I think it will be good to lead the users in case of non-dvc-repo errors. This page describes the different ways they can prepare their repo for viewing in Studio, and has a link to the DVC doc. Let me know what you think.

shcheklein replied:
I think it's complicated still, but it's fine to start here. Later we could use a page similar to this https://dvc-org-studio-docs-v2-kfehbw2.herokuapp.com/doc/user-guide/troubleshooting 

And a mechanism with redirects - https://error.dvc.org/lock-issue

tapadipti replied:
Added to #2515

tapadipti commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+  to the sub-directory that contains the DVC repo for which you want to create
+  the view.
+
+- **Data remotes:** If you need to set up DVC data remotes for your view, you
```
@shcheklein Let me know if you think we need to add more details regarding data remotes setup.

shcheklein replied:
it would be great to mention:

- we dont't read/access data
- we use it only to read plots/metrics files if they are not saved into Git. And people should use it only if they have this situation. They will see a specific message in their View.

tapadipti replied:
Adding this: "Note that DVC Studio uses the credentials only to read plots/metrics files if they are not saved into Git. It does not access any other data in your remote storage. And you do not need to provide the credentials if any DVC data remote in not used in your Git repository."

casperdcl commented on content/docs/studio/create-view.md:
```diff
@@ -1,67 +1,71 @@
 
-4. Select the Git repository for which you want to create a view. For this,
-   hover over the required repository and click `Connect`.
+   ![](https://static.iterative.ai/img/studio/select_repo_v2.png)
```
I don't think we need any of these `_v2` things

- we'd initially used `static.iterative.ai` because the images were WIP and we knew we'd change them
- now that they're finalised (I assume? #2469) we can put them in this repo as usual

CC @shcheklein related #93 <- #1115

tapadipti replied:
The reason I used _v2 is because I needed the images uploaded for testing locally. But didn't want to mess up the prod versions until the PR was reviewed and ready.

tapadipti replied:
Added to https://github.com/iterative/dvc.org/issues/2515 for removing _v2 from the image filenames (after launch).

casperdcl commented on content/docs/studio/create-view.md:
```diff
@@ -1,67 +1,71 @@
-5. Once you click on `Connect`, you will be able to specify advanced settings
-   for the connection.
-   ![](https://static.iterative.ai/img/studio/view_settings.png)
+4. You will be able to specify additional settings for the connection.
```
other steps used imperative style so for consistency:

```suggestion
4. Specify additional connection settings if required.
```

casperdcl commented on content/docs/studio/create-view.md:
```diff
@@ -1,67 +1,71 @@
+> [View Settings](/doc/studio/view-settings) section for more details.
 
-![](https://static.iterative.ai/img/studio/view_added.png)
+You should now see that a view has been added in your dashboard.
```
```suggestion
You should now see that a View has been added in your dashboard.
```

casperdcl commented on content/docs/studio/create-view.md:
```diff
@@ -1,67 +1,71 @@
-  [monorepo](https://en.wikipedia.org/wiki/Monorepo), then specify the full path
-  to the root directory of the project that you want to connect.
+You can change a shared (public) view and make it private by turning off
+`Share to web`.
```
```suggestion
`Share to Web`.
```

casperdcl replied:
btw is "share to web" a better label than simply "share?"

"to web" implies there are other options ("by email?")

tapadipti replied:
`Share to web` is how the link appears on Studio. So, not changing it here. I think `Publish to web` may be a better option. But it needs to be changed in Studio first, before changing in the docs.

shcheklein replied:
yep, create a ticket on in the `iterative/viewer` please. Good cath @casperdcl .

tapadipti replied:
Created https://github.com/iterative/viewer/issues/2098

casperdcl commented on content/docs/studio/create-view.md:
```diff
@@ -1,67 +1,71 @@
-  Enter the full file path, and specify whether the file is for `Metrics` or
-  `Parameters`.
+If you are using a GitHub account, you will have to install the DVC Studio
+GitHub app on your GitHub repositories / organizations that you want to use with
```
```suggestion
GitHub app on the GitHub repositories/organizations that you want to use with
```

casperdcl commented on content/docs/studio/create-view.md:
```diff
@@ -1,67 +1,71 @@
+to install and authorize the DVC Studio GitHub app.
+
+> Note that you do not have to authorize DVC Studio on all the repositories in
+> your GitHub organization. You can limit access to only those repositories
+> which you want to use with DVC Studio.
```
```suggestion
> your GitHub organization. You can limit access to only repositories that
> you want to use with DVC Studio.
```

casperdcl commented on content/docs/studio/get-started.md:
```diff
@@ -1,31 +1,34 @@
+repositories or you can add custom files with the required data. Refer to
+[View settings](/doc/studio/view-settings) to understand the different ways in
+which you can prepare your Git repository for use with DVC Studio. Using this
+data, DVC Studio creates a summary view. This includes tables of all the metrics
```
not sure about capitalisation consistency (summary view/summary View/Summary View)

tapadipti replied:
Since this is not a keyword, I guess we can leave it at 'summary view'? What do you think?

casperdcl replied:
idk, I just noticed you'd been using capital "View" earlier so I assume it should be the same here.

shcheklein replied:
` This includes tables` - why tables (plural?). To my mind view == table (with some filters applied).

casperdcl commented on content/docs/studio/index.md:
```diff
@@ -1,25 +1,26 @@
 
-- **Run [CI/CD](https://en.wikipedia.org/wiki/CI/CD) for your ML projects on
-  cloud resources of your choice without any new tools.**
+- Run [CI/CD](https://en.wikipedia.org/wiki/CI/CD) for your ML projects on cloud
```
is this related to CML or is it a Studio feature?

tapadipti replied:
Studio, by virtue of CML. That is why the title for this section is "Use DVC, CML and Studio to:"

casperdcl commented on content/docs/studio/run-experiments.md:
```diff
@@ -5,24 +5,55 @@ CI/CD set-up is used to run the experiments. For instance,
+> Note that due to access restrictions, you cannot run experiments on the demo
+> view (`example-get-started`) that is provided to you by default. Once you
+> create views for your ML project repositories, you can follow the instructions
+> given below and run experiments directly from DVC Studio.
```
```suggestion
> given below to run experiments directly from DVC Studio.
```

casperdcl commented on content/docs/studio/run-experiments.md:
```diff
@@ -5,24 +5,55 @@ CI/CD set-up is used to run the experiments. For instance,
+
+![](https://static.iterative.ai/img/studio/cml_commit_v2.png)
+
+At this point, the new experiment appears in the view table. If you check in
+your Git repository, you will find the Git commit that you just made. If you had
+committed to a new branch, a new pull request will also have been created from
+the new branch to the base branch.
```
```suggestion
At this point, the new experiment appears in the view table. If you just
committed to a new branch, then a new pull request will also have been created
from the new branch to the base branch.
```

casperdcl commented on content/docs/studio/run-experiments.md:
```diff
@@ -5,24 +5,55 @@ CI/CD set-up is used to run the experiments. For instance,
+committed to a new branch, a new pull request will also have been created from
+the new branch to the base branch.
+
+If your project is integrated with a CI/CD setup (eg, GitHub Actions), the CI/CD
```
```suggestion
If your project is integrated with a CI/CD setup (e.g. GitHub Actions), the CI/CD
```

casperdcl commented on content/docs/studio/run-experiments.md:
```diff
@@ -5,24 +5,55 @@ CI/CD set-up is used to run the experiments. For instance,
+setup will get invoked. If this setup includes a model training process, it will
+be triggered, which means that your ML experiment will run automatically. The
+model training can happen on any cloud or Kubernetes. For more details on how to
+set up CI/CD pipelines for your ML project, refer to
+[CML](https://dvc.org/doc/cml) .
```
```suggestion
set up CI/CD pipelines for your ML project, refer to [CML](https://cml.dev).
```

casperdcl commented on content/docs/studio/teams.md:
```diff
@@ -43,19 +28,35 @@ collaborators by accessing team settings later.
 
+## Roles
+
+Team members can have different roles.
```
maybe best to tabulate this?

tapadipti replied:
Ok, moving this to a follow-up ticket.

tapadipti replied:
Added to #2515

shcheklein replied:
Agreed - it should be a matrix

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+the Git repository and does not reference remote/cloud storage, then you can
+successfully visualize it without specifying additional settings.
+
+In other cases, you could be trying to create views from non-DVC repositories.
+Or, your project may be in a sub-directory inside a monorepo. Or, the data to be
+visualized could be in custom files in your repository or even in remote / cloud
+storages. In all these scenarios, you will need to specify advanced settings so
+that DVC Studio is able to access the data required for visualization. These
+scenarios have been explained in details below.
```
```suggestion
Alternatively, you could create views from:

- Non-DVC repositories
- Project sub-directories in a monorepo
- Custom files in your repository or remote/cloud storage

In each of these scenarios, you will need to specify advanced settings for DVC
Studio to be able to access the data required for visualization. Details are
given below.
```

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+
+## Non-DVC repositories
+
+DVC Studio creates views by identifying datasets, metrics and hyperparameters
```
```suggestion
DVC Studio creates Views by identifying datasets, metrics and hyperparameters
```

tapadipti replied:
Not sure if the plural 'Views' is also a keyword for us. I think non-cap views would be ok. What do you think?

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+1. **Set up DVC repositories**: You can use [DVC](https://dvc.org/) and Git to
+   version your code, data and models all within your Git repositories. By using
+   DVC, you can be sure not to bloat your repositories with large volumes of
+   data or huge models. These large assets reside in cloud or other remote
```
```suggestion
   data or huge models. These large assets reside in the cloud or other remote
```

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+   version your code, data and models all within your Git repositories. By using
+   DVC, you can be sure not to bloat your repositories with large volumes of
+   data or huge models. These large assets reside in cloud or other remote
+   storage locations. You will simply track their version info in Git.
```
do we need to explain what DVC is over here? In which case there's probably already some other paragraph from elsewhere in the docs which we can copy-paste or link to.

tapadipti replied:
Yes. Will add.

tapadipti replied:
Added to #2515

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+   repository. You can then connect to this DVC repository and create a view as
+   described [earlier](/doc/studio/create-view). DVC Studio automatically
+   detects metrics, plots, and hyperparameters files specified in the project's
+   `dvc.yaml`. And every time you push a new commit to this DVC repository, your
+   view will reflect the new changes.
```
```suggestion
   `dvc.yaml`. Each time you push a commit to this DVC repository, your view
   will reflect the new changes.
```

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+   view will reflect the new changes.
+
+2. **Specify custom files with your metrics and parameters**: If you are working
+   with a non-DVC repository, you can still create views for it if your
+   repository contains CSV, JSON or YAML files with the metrics and
+   hyperparameters that you want to visualize. To enable DVC Studio to visualize
+   such custom data, simply
+   [specify the custom files](#specifying-view-settings) with your metrics and
+   parameters, and DVC Studio will efficiently generate tables and plots for
+   your custom input. For instance, if you have an ML project for which you
+   generate and save metrics either manually or using some ML tracking tools,
+   then you can create a view for this project by specifying the file (within
+   your Git repo) which contains your saved metrics.
```
```suggestion
   with a non-DVC repository, you can still create views for it provided that
   metrics and hyperparameters are stored in CSV, JSON or YAML files. To
   visualize such custom data, simply
   [specify the custom files](#specifying-view-settings) to use, and DVC Studio
   will efficiently generate tables and plots for your custom input. For instance,
   if you have an ML project for which you generate and save metrics either manually
   or using some ML tracking tools, then you can create a view for this project by
   specifying the file (within your Git repo) which contains your saved metrics.
```

casperdcl replied:
or even briefer:

```suggestion
   with a non-DVC repository, you can still create views for it provided that
   metrics and hyperparameters are stored in CSV, JSON or YAML files. To
   visualize such custom data, simply
   [specify the custom files](#specifying-view-settings) to use, and DVC Studio
   will efficiently generate tables and plots for your custom input.
```

tapadipti replied:
I think the extra explanation may help some people. So, accepting your earlier suggestion.

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+will need to specify the full path to the sub-directory that you want to use
+with your view.
+
+## Data remotes (cloud / remote storage)
```
```suggestion
## Data remotes (cloud/remote storage)
```

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+## Data remotes (cloud / remote storage)
+
+The metrics and parameters that you want to include in the view may also be
+present in a data remote(cloud storage or another location outside of the Git
```
```suggestion
present in a data remote (cloud storage or another location outside the Git
```

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+  the view.
+
+- **Data remotes:** If you need to set up DVC data remotes for your view, you
+  will need to do it after your view has been created. So, first create your
```
```suggestion
  will need to do it after your view has been created. First, create your
```

casperdcl commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,92 @@
+  View settings. Open the `Data remotes / cloud storage credentials` section.
+  The data remotes that are used in your DVC repo will be listed. Now, click on
+  `Add new credentials`. In the form that opens up, select the provider (Amazon
+  S3, GCP, etc.). depending on the provider, you will be asked for more details
```
```suggestion
  S3, GCP, etc.). Depending on the provider, you will be asked for more details
```

casperdcl commented on src/components/LayoutHeader/alert.tsx:
```diff
@@ -7,11 +7,11 @@ import styles from './styles.module.css'
-    <span role="img" aria-label="shield">
-      🛡
+    <span role="img" aria-label="studio">
+      <img src="img/studio-icon.svg" alt=""></img>
```
not sure if this needs to be `className` from `styled.module.css` or even an SVG `ReactComponent` CC @julieg18 @rogermparent

tapadipti replied:
Added to #2515 for fixing if needed.

rogermparent replied:
While it looks like this works totally fine, the first iteration of this alert bar [did use a ReactComponent import](https://github.com/iterative/dvc.org/blob/809ac154ef02a8d9847432ae8ffd5bca666828f0/src/components/LayoutHeader/alert.tsx) to display a GitHub icon svg- everything else has been emojis/unicode IIRC.

An SVGR `ReactComponent` import seems like it would be more optimal, the difference is probably negligible to the point where is `ReactComponent` causes any problems (like with colors, for example, which didn't come into play on the GH svg), it's fine to stick with `img`.

tapadipti replied:
@rogermparent There was some issue when we used `img`. We were getting Cloudflare alert that the page was offline. It was not breaking all the time, but Ivan noticed it when he was trying to load the page. We replaced the svg img with 🚀 in this [PR](https://github.com/iterative/dvc.org/pull/2532/files) and the alerts went away (page stopped breaking). Any idea why that may have happened and how we can use images if needed?

shcheklein replied:
@tapadipti let's please create a proper issue/bug with the screenshot, what we have done to fix, etc

jorgeorpinel replied:
https://github.com/iterative/dvc.org/issues/2540

tapadipti commented on content/docs/start/index.md:
```diff
@@ -76,3 +76,8 @@ by one in the next few pages:
+
+**New!** Once you set up your DVC repository, you can also interact with it
+using DVC Studio, the online UI for DVC.
+[Here's a demo](https://studio.iterative.ai/team/Iterative/views/example-get-started-zde16i6c4g)
```
This will remain accessible.
@shcheklein can u pls confirm in case I am wrong.

shcheklein replied:
it's not a commit hash - it's a unique `view id`. It will stay stable until we remove that view.

tapadipti commented:
@casperdcl thanks for your review. I've addressed your comments, and created a new ticket with pending issues. https://github.com/iterative/dvc.org/issues/2515

casperdcl reviewed (approved):
approving to unblock merge (follow-up #2515)

shcheklein commented on content/docs/studio/index.md:
```diff
@@ -1,25 +1,26 @@
 
 [`DVC Studio`](https://studio.iterative.ai/) is a comprehensive and interactive
-collaboration tool for your Machine Learning projects.
+web application to manage and collaborate on your Machine Learning projects
```
why don't use the same simple language as we do in blog here:

```
DVC Studio is a web application that you can
[access online](https://studio.iterative.ai/) or even host on-prem. It works
with the data, metrics and hyperparameters that you add to your ML project
repositories.
```

tapadipti replied:
Changed.

shcheklein commented on content/docs/studio/get-started.md:
```diff
@@ -1,31 +1,38 @@
 ## What is a view?
 
 A _view_ is an interactive representation of the experiments run in your ML
 project. DVC Studio automatically identifies datasets, metrics and
```
I think `automatically` is not the best term here. It doesn't do anything automatically. It parses Git history and specific metafiles (or files you show to it). Automatically sounds like some magic is happening.

tapadipti replied:
removed

casperdcl replied:
"intelligently?"

shcheklein commented on content/docs/studio/get-started.md:
```diff
@@ -1,31 +1,38 @@
 
 Here, we will walk you through a tutorial to use DVC Studio for collaboration on
-your ML projects. You will need access to a Github, Gitlab or Bitbucket account
+your ML projects. You will need access to a GitHub, GitLab or Bitbucket account
```
in the "DVC Studio creates views" view is not defined yet. Better to use some human readable terms - e.g. table with experiments? etc

tapadipti replied:
Will add a link from 'views' to the 'What is a view?' section on that page. Works?

shcheklein commented on content/docs/studio/get-started.md:
```diff
@@ -1,31 +1,38 @@
-required data (more on this later). Using this data, DVC Studio creates a
-summary view. This includes tables of all the metrics across all your
-experiments. You can also generate plots and compare experiments here.
+hyperparameters in your ML experiments. The data can either be in DVC
```
data has a very specific meaning in ML, here the use of `data` is misleading

tapadipti replied:
changing 'The data' to 'These details'

shcheklein replied:
pls check other places

how about Values ...?

casperdcl replied:
"This info?"

shcheklein commented on content/docs/studio/view-settings.md:
```diff
@@ -0,0 +1,100 @@
+hyperparameters) are stored in your Git repositories as CSV, JSON or YAML files.
+You can add these details to your Git repositories in two ways:
+
+1. **Set up DVC repositories**: You can use [DVC](https://dvc.org/) and Git to
```
DVC is not only about not bloating Git repo. It's about pipelines, connecting then with CML, etc. We should be explaining these benefits better I think

tapadipti replied:
We should have a separate section/paragraph on DVC and its benefits, and link it to the DVC doc. For now, I'm adding this to the current paragraph: "DVC also
   enables you to
   [share your data and model files](/doc/use-cases/sharing-data-and-model-files),
   [create data registries](/doc/use-cases/data-registries),
   [create data pipelines](/doc/start/data-pipelines), connect them with
   [CML](/doc/cml) for CI/CD in machine learning, and so on. Find more about the
   features and benefits of DVC [here](/doc/start)."

shcheklein replied:
ideally we should zoom out and summarize high level benefits here (e.g. pipelines == reproducibility).

One way can be - DVC goes beyond that. It lets you organize your project in a reproducible way, it gives you way to save your experiments holistically as Git commits - data, models, metrics, params ... You could use it to organize manage your data using Git flow (data registries) ... 

something like this ...

shcheklein reviewed (approved):
Approving this, so that it can be merge before the release
